### ISSUE TITLE
svg files sometimes display incorrect colors in emacs-plus@29

### ISSUE THREAD

**Issue by ilyaw39** [d12frosted/homebrew-emacs-plus#522]
When I use `dvisvgm` to render LaTeX formulas, with extra font packages such as `txfonts`, the images are displayed incorrectly.

Environment:

```
(setq org-latex-packages-alist
      '(("" "txfonts" t)))

(setq org-preview-latex-default-process 'dvisvgm)
```

I first considered that Emacs got the wrong `:foreground` color when generating the LaTeX overlays.

But the following evidence I noticed suggests that it is more likely that Emacs just didn't display `svg` file correctly, and has nothing to do with LaTeX's rendering and displaying process:
![](https://i.imgur.com/jetLwdb.png)
And I also tested the `svg` file's source code on web, and got the correct display:

![](https://i.imgur.com/JObHzoT.png)

Additionally, both the rendered LaTeX overlay and the original svg file can be displayed correctly in `emacs-mac`. So I consider this may have something to do with `librsvg` (I got `librsvg` installed with `homebrew` and kept it up-to-date).

**Comment by d12frosted** [comment]
Hey @ilyaw39

Thanks for the detailed report. Not sure if I can do anything about as Emacs+ is just GNU Emacs formula, and this is how GNU Emacs works. On the contrary, `emacs-mac` is a fork with lots of modifications (including the rendering part).

BTW you might find #258 interesting. My point still stands: if someone finds a good and minimal fix for this issue, would gladly accept the patch.

**Comment by ilyaw39** [comment]
> Hey @ilyaw39
> 
> Thanks for the detailed report. Not sure if I can do anything about as Emacs+ is just GNU Emacs formula, and this is how GNU Emacs works. On the contrary, `emacs-mac` is a fork with lots of modifications (including the rendering part).
> 
> BTW you might find #258 interesting. My point still stands: if someone finds a good and minimal fix for this issue, would gladly accept the patch.

Thanks for the instructions ☺️. I have now found a very likely cause of the error and fixed this issue locally.

The error should be related to my local Mac system and `homebrew`. Because I had been using `macOS 13 beta` before, and `homebrew` did not adapt to macOS version at that time. After installing the `stable` version of macOS 13 released today, I updated `homebrew` and reinstalled `librsvg`, and noticed the words `arm64_ventura.bottle` during the installation process:

````
==> Pouring librsvg--2.54.5.arm64_ventura.bottle.tar.gz
````

Then I reinstalled `emacs-plus@29`, and luckily found that the original problem has been solved.

![](https://i.imgur.com/ldFDYHb.png)

**Comment by d12frosted** [comment]
Oh wow, happy to hear. And thank you for sharing your workaround.

**Comment by ilyaw39** [comment]
Problems are now solved in [525#issuecomment-1318055049](https://github.com/d12frosted/homebrew-emacs-plus/issues/525#issuecomment-1318055049). I'll close this issue unless new issues related are reported.
